### DISCUSSION TITLE
2021 Cloudlog Development (Thoughts Needed)

### DISCUSSION THREAD

**Discussion by magicbug** [magicbug/Cloudlog#789] [General]
First off Happy New Year all from myself, pleased to say although 2020 was a bit of a disaster I'm all OK up in the NE of Scotland, however with lockdowns it certainly gave more time for Cloudlog.

The biggest thing with Cloudlog that I noticed was a tremendous amount of community participation, I never thought that Cloudlog would be anything more than a tool myself and a few friends would use, but that's clearly not the case and I want to thank @AndreasK79 for all this work with the analytics features, @dg9vh for the improved modes system and @poll-busily who clearly hates my typos and awful grammar to fix it all 👍🏻 

On top of that, I managed to get quite a few internal bugs fixed and polished the interface and the main feature I've wanted in cloudlog in the form of LoTW sync removing the need for tqsl.

That brings us to 2021 and what perhaps needs to go on the roadmap for development, to give me and anyone else who likes to submit PRs some ideas, I still have the printable labels system to finish off, a lot of the codes in place but still needs some love before it's all ready for release, I know a few are waiting on this including me.

I think the other part that needs a serious overall is the authentication system for signup and roles and to start allowing a better multi-user environment, as with anything this is actually a bigger job than some might think but its always in the back of my mind along with allowing linking Station Profiles into an overall "logbook".

On top of that Global Options need to be used more within Cloudlog completely removing the need to edit the config files, think at the moment it's just Callbook logins that need to be moved across.

A few other bits for integration I want to look at is

- Using Web Sockets for Radio interfaces, this should mean fewer hits against the database but allow for perhaps more options like CW macros, and two-way control of settings.
- Create some scripts to integrate Macdoppler and GPredict into Cloudlog like SatPC32 (I actually don't have access to a mac to do Macdoppler but I can do GPredict)
- Add all the code in Cloudlog so that translation files can be added there is a few bits that are already done but all the views need to be attacked.

Happy to hear peoples thoughts, to see what we can do this year.

Peter, 2M0SQL

**Comment by ghost** [comment]
Happy New Year everyone. I don't have any strong feelings about the priorities in terms of new major features; as someone who hasn't been involved in amateur radio for that long (<1 year) I'm still taking stock. I enjoy using Cloudlog and I think it's definitely moving in the right direction. I do like the ideas of reworking the user roles and of moving application config to the Global Options.

As you've gathered, I like to work on polishing and refining things. I think that's really important for getting more people to use Cloudlog and encouraging people to contribute to the project.

Here are a few areas I'd like to tackle sometime, in no particular order:
- Security review (this is my day job and if I find the time I'll have a look at Cloudlog in depth)
- Add documentation and expand what's there
  - Fully document the API (probably ties in with the auth/roles work)
- Help expand the scope of the translations
- Improve the out-of-box / first run experience
- Tidy some of the source files
- More UI tweaks
  - Hide/disable menu items if your user role doesn't allow access?
  - Enable/disable bands, similar to enabling/disabling modes (I only use 20m,2m,70cm and don't want to see all of them)
  - Cache more things in local storage / session (e.g. last used mode)

If anyone wants to collaborate on those then let me know.

**Reply by magicbug** [reply]
@poll-busily everything you list will please me, security review very much so especially having a different set of eyes on it, agree might fall in with roles / user system improvements, perhaps we can start a new thread on this think first setup might also come under this, I do have some new code and displays in a branch that I want to use to replace the sign up system.

UI tweaks always needed, as you say again falls a bit into Roles too, for example Admins don't need global options. bands perhaps needs similar to Modes, I'd also like to see an option for both for a default option so if you use CW its the one that shows on load, should have similar for a band.

As you probably spotted, starting to use localStorage with contest page, still plenty we can do on this, again perhaps a good separate thead.

Docs YES! trying, but first to admit I'm lazy and rubbish when it comes to such things, but this all helps new users.

**Comment by ClaudeMa** [comment]
Hello

If you need a french translator i can do it
73
Claude

Le dim. 3 janv. 2021 à 15:15, Paul Beesley <notifications@github.com> a
écrit :

**Reply by magicbug** [reply]
@ClaudeMa Fantastic, it's still a work in progress but see https://github.com/magicbug/Cloudlog/discussions/795

**Comment by AndreasK79** [comment]
Happy new Year Peter and everyone else. I'm always happy to help out where I can.

The log has grown a lot the past year, that's for sure. I still think it needs more polishing here and there, but it's getting there.

The things I'm thinking should be pretty high up on the todo list are:

- Translation
- Multi-user / multi-logbook
- More and better documentation. I was thinking about adding a F.A.Q. section, as I see many of the same questions popping up.
- Improve the out-of-box / first run experience
- Move away from the config-file, and instead use the database to expose settings in the GUI.

This is not necessarily my list of things, as I try to pitch in wherever I can.

Should we also somehow try to coordinate stuff we work on? I know I've just picked stuff here and there, but that is not good if several people are working on the same stuff. 

Andreas, LA8AJA

**Comment by daccle** [comment]
Happy new year everybody!

I am back on track and will get back to spend time on the development of Cloudlog. During my break I've used Cloudlog quite a bit and this was a good exercise to understand how things are meant to be working. There are many tiny bits which can be improved to improve the general user experience and many small things to document. I will try to be useful on these parts.

Besides I working on improving the installation process of Cloudlog. I try to test each of my PRs on a clean install and I am using Docker to run a test environment. Once the setup process is more automated and stable I will share. So that testing for devs becomes easier.

I am quite used to review code and to code in a team. So, if you want feedback for some PR or would like to partner on some PR please feel free to grab me.

Daniel DC2ZP

**Comment by myzinsky** [comment]
I still would like to work on a two-way sync. Since I worked a lot with directus in the last months, I'm already very spoiled with REST-APIs :-) Something like that would be great for CL too!

73, DL9MJ

**Comment by oh1jw** [comment]
Any news about Macdoppler UDP script? 
What you guys think about landing page (option)?

**Reply by magicbug** [reply]
I actually haven't gotten far with it at all, lack of being a Mac user doesn't help.

Landing page in what sense?

**Reply by oh1jw** [reply]
> I actually haven't gotten far with it at all, lack of being a Mac user doesn't help.
>
I understood that you and M0LTE have started something together.. 

> Landing page in what sense?
>
In that sense, that you could edit the "landing page info" for index.php for non registered viewers. 
Like putting some info about your station or upcoming activations or your full life story with every bits and bobs. 
Or totally remove the qso infos etc..

**Reply by dfbarnes** [reply]
I like the landing page idea / customizable options for index.php.

**Reply by magicbug** [reply]
Landing page idea could work, @oh1jw I'd suggest creating a separate thread on it and tagging as Idea, the landing page things need to change a bit with auth changes so this would work nicely into all that

**Reply by oh1jw** [reply]
Landing page discussion will continue here: https://github.com/magicbug/Cloudlog/discussions/1045#discussion-3383435

**Comment by kc9mne** [comment]
I would like to see offline ability, I use CloudLog for field day running on a server everyone just needs a browser no software hassles no cares if the internet goes down its all local. I would like to remove callsign lookup dependencies on 3rd party sites I know I can grab the weekly fcc DB from https://www.fcc.gov/uls/transactions/daily-weekly?job=transaction&page=weekly with FTP.   This way I can have a PI or other low power sbc onsite running all the logging as well as a rocket chat instance for chat between operator stations. (a lot of times we have a bunch of tents separated )

**Reply by magicbug** [reply]
I have been wondering about this about having an internal callbook, obviously would create much larger requirement for database, but all it would require is an new callbook plugin and some code to handle the rest.

You'd need to also either turn off the maps or run a local openstreetmap system.

**Reply by kc9mne** [reply]
I was looking at that or just having a high level set of tiles so you can get a general overview of a contacts location it really does not need to get more detailed than a general area.
